### DISCUSSION TITLE
Question: How can you specify a target to execute on an external cake script?

### DISCUSSION THREAD

**Discussion by Kimserey** [cake-build/cake#2856] [Q&A] [answered]

**Comment by devlead** [comment, marked as answer]
If you're talking about [CakeExecuteScript](https://cakebuild.net/api/Cake.Common.Tools.Cake/CakeAliases/5FCD85B6) then you can use the [Arguments](https://cakebuild.net/api/Cake.Common.Tools.Cake/CakeSettings/61A4B9B2) property on its [CakeSettings](https://cakebuild.net/api/Cake.Common.Tools.Cake/CakeSettings/) parameter.

**Reply by Kimserey** [reply]
Thanks @devlead , this was what I was looking for.

I was calling the external cake script with the argument `target="abc"` 

```
CakeExecuteScript("./build1.cake", new CakeSettings{ Arguments = new Dictionary<string, string>() { { "target", "MyTarget" } } })
```

but I was missing the argument itself on the external cake script:

```
var target = Argument("target", "Default");

// targets here...


RunTarget(target);
```
